outputs.tf

```diff
@@ -30,3 +35,41 @@ output "nat_route_id" {
   description = "id of VCN NAT gateway route table"
   value       = join(",", oci_core_route_table.nat.*.id)
 }
+

```

Comment:
I like the idea but add it in another PR please. Also, when you add them, provide an example of how you would use them in the examples/.

Also, just curious, what other attributes you're looking to reuse?

Reply:
OK I rollback this a propose a separate change.

Explanation:

When wrapping a resource behind a module, the most common problem is that you cannot interact with it as you would with the original resources, or let's say you can interact with the module inner resources as much as your Inputs and Ouputs permits.

This output gives you access to all existing attributes, and that's key for reusability with module composition and when using [remote state data source](https://www.terraform.io/docs/language/state/remote-state-data.html).

Some examples that could be interesting on a vcn: `vcn_domain_name`, `display_name`, the default resources ids (default SL, default RT, default DHCP Options), `time_created`.

All of this information comes for free and little effort. VCN is a fundamental module: we don't know yet what will be built with it, so the tool should allow for great level of flexibility.

Last one that is really important: any new attribute as the provider evolves is added, and on the long run this means low maintenance for us.

Reply:
All right but send in another PR please.

Reply:
Issue #34 has been created for this subject.